packages/hub/services/nonce-lock.ts

```diff
@@ -0,0 +1,79 @@
+import { inject } from '@cardstack/di';
+import BN from 'bn.js';
+import config from 'config';
+import { addSeconds } from 'date-fns';
+import shortUUID from 'short-uuid';
+import { nowUtc } from '../utils/dates';
+
+export default class NonceLock {
+  prismaManager = inject('prisma-manager', { as: 'prismaManager' });
+  ethersProvider = inject('ethers-provider', { as: 'ethersProvider' });
+
+  private async getNonce(accountAddress: string, chainId: number) {
+    let prisma = await this.prismaManager.getClient();
+    return await prisma.$transaction(
+      async (tx) => {
+        await tx.$executeRaw`SELECT * FROM account_nonces WHERE account_address = ${accountAddress} AND chain_id = ${chainId} FOR UPDATE`;
+
+        let accountNonce = await tx.accountNonce.findUnique({
+          where: { accountAddress_chainId: { accountAddress, chainId } },
+        });
+        let one = new BN(1);
+        let nextNonce = accountNonce?.nonce ? new BN(accountNonce.nonce.toString()).add(one) : new BN(0);
```

Comment:
we should probably ask ethers.js for the current nonce for the EOA and add one to that right? after all we are trying to reflect what is happening in ethereum--ethereum is the source of truth not us. if the nonce we get back is one that is already in the DB then we know something is wrong (probably related to my other comments around what happens when a txn reverts and a nonce is not advanced).

Reply:
oh wait never min. the API i'm thinking of is from web3.js. in web3 you can ask an EOA's current nonce (`web3.eth.getTransactionCount(address)`), but in ethers.js, the transaction count is only known _after_ you send the txn, which you determine on line 27 below.